notifications/notifier.go

```diff
@@ -0,0 +1,170 @@
+package notifications
+
+import (
+	"context"
+	"io"
+	"net/http"
+
+	logging "github.com/ipfs/go-log/v2"
+	"github.com/textileio/powergate/v2/ffs"
+)
+
+var (
+	log = logging.Logger("notifier")
+)
+
+type Notifier interface {
+	RegisterJob(jobId ffs.JobID, configs []*ffs.NotificationConfig)
+	NotifyJobUpdates(job JobUpdates)
+}
+
+type notifier struct {
+	ctx           context.Context
+	configs       *configStore
+	updates       chan JobUpdates
+	toDelete      chan ffs.JobID
+	notifications chan *notification
+}
```

Comment:
Very nice design of the notifier with workers. I like how the code puts the job updates through a funnel of logic to figure out exactly who to notify. Cool stuff.

Reply:
thank you :)